lib/src/csv_parser.cpp

```diff
@@ -103,8 +105,30 @@ Problem CsvParser::load_input() const {
     return problem;
 }
 
-void CsvParser::save_output(const Solution& sln) const {
-    UNUSED(sln)
+void CsvParser::save_output(const Problem& prb, const Solution& sln) const {
+	std::ofstream outfile("solution.csv");
+
+	outfile << "Route;" << "Vehicle;" << "Customer;" << "Arrive;" << "Begin;" << "End;" << "Leave;" << "Distance from previous;" << "Depot distance;" << std::endl;
+
+	char del = ';';
+
+	for (size_t i = 0; i < sln.routes.size(); ++i) {
+		for (size_t j = 0; j < sln.routes[i].second.size(); ++j) {
```

Comment:
~~~c++
for (size_t i = 0; i < sln.routes.size(); ++i) {
  const auto& route = sln.routes[i]; // name `route` is optional. rename as you see fit
  for (size_t j = 0; j < route.second; ++j) {
    // do stuff
  }
}
~~~